provider/factory.go

```diff
@@ -0,0 +1,26 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+	"github.com/hashicorp/terraform-plugin-mux/tf5muxserver"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// ProtoV5ProviderServerFactory returns a muxed terraform-plugin-go protocol v5 provider factory function.
+// The primary (Plugin SDK) provider server is also returned (useful for testing).
+func ProtoV5ProviderServerFactory(ctx context.Context, version string) (func() tfprotov5.ProviderServer, *schema.Provider, error) {
+	primary := New(version)
```

Comment:
`primary` doesn't seem like the right name moving forward. Once we have a second provider based on the plugin framework then this would at best be `secondary`. 

Perhaps something like `sdk2Provider` or something similar instead?